addons/sale_stock/models/sale_order.py

```diff
@@ -15,9 +15,7 @@ class SaleOrder(models.Model):
 
     @api.model
     def _default_warehouse_id(self):
-        company = self.env.company.id
-        warehouse_ids = self.env['stock.warehouse'].search([('company_id', '=', company)], limit=1)
-        return warehouse_ids
+        return self.user_id._get_default_warehouse_id()
```

Comment:
self.user_id will always be an empty recordset here, because self is an empty recordset in default method calls ;).
--> `self.env.user._get_default_warehouse_id()`